firefox-ios/Client/Telemetry/TelemetryWrapper.swift

```diff
@@ -104,6 +104,14 @@ class TelemetryWrapper: TelemetryWrapperProtocol, FeatureFlaggable {
             GleanMetrics.LegacyIds.clientId.set(uuid)
         }
 
+        // Set or generate profile id used for usage reporting
+        if let uuidString = profile.prefs.stringForKey(PrefsKeys.Usage.profileId), let uuid = UUID(uuidString: uuidString) {
+            GleanMetrics.Usage.profileId.set(uuid)
+        } else {
+            let uuid = GleanMetrics.Usage.profileId.generateAndSet()
+            profile.prefs.setString(uuid.uuidString, forKey: PrefsKeys.Usage.profileId)
+        }
```

Comment:
This should only happen if `sendUsageData == true`.
If `false` we should also set the canary value `"beefbeef-beef-beef-beef-beeefbeefbee"` (as above)